docs/endpoint.md

```diff
@@ -5,8 +5,103 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
 The encrypted message would be sent to Tangle with a streaming message channel API. The streaming message channel API would ensure the order of messages in the channel. The user who wants to fetch/send message to Tangle needs to provide `data_id`, `key` and `protocol` to identify a specific message.
 A message sent by endpoint needs to be encrypted locally, which avoids message being peeked and modified. 
 
-# How to use
+# How to build endpoint
+
+## Setup sierra dev environment 
+The endpoint uses the sierra framework as based runtime system. Needed to setup the sierra dev environment to build endpoint as specific target.
+
+### How to build wp77xx
+
+Leaf is a workspace manager that will download, install and configure the required software packages for a Legato development environment.
+
+```
+$ wget https://downloads.sierrawireless.com/tools/leaf/leaf_latest.deb -O /tmp/leaf_latest.deb && sudo apt install /tmp/leaf_latest.deb
+```
+
+Setup the wp77xx target profile
+```
+leaf setup legato-latest -p swi-wp77_3.3.0
+```
+
+Finally, make the wp77xx endpoint target.
+```
+$ make TARGET=wp77xx legato // build endpoint as wp77xx target
+$ make ENDPOINT_TEST_ENABLE=true TARGET=wp77xx legato // build endpoint as wp77xx target in test mode
+```
```

Comment:
I think we should clarify that the downloaded `tangle-accelerator` directory should be located at the same level or subdirectory of the workspace directory created by `leaf setup legato-latest -p swi-wp77_3.3.0`.